Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -531,117 +522,144 @@ section ModelsWithJ
 
 variable (R)
 
-/-- When 3 is invertible, $Y^2 + Y = X^3$ is an elliptic curve.
-It is of j-invariant 0 (see `EllipticCurve.ofJ0_j`). -/
-def ofJ0 [Invertible (3 : R)] : EllipticCurve R :=
-  have := invertibleNeg (3 ^ 3 : R)
-  ⟨WeierstrassCurve.ofJ0 R, unitOfInvertible (-3 ^ 3 : R),
-    by rw [val_unitOfInvertible, WeierstrassCurve.ofJ0_Δ R]; norm_num1⟩
+-- TODO: change to `[IsUnit ...]` once #17458 is merged
```

Comment:
Shall we wait until #17458 is merged before continuing on this PR?

Reply:
Sorry, I just got back to this. Once that is merged, the suggestions I have below will be drastically different.